docs/elasticsearch-spec.asciidoc

```diff
@@ -101,42 +144,94 @@ $ openssl req -x509 -newkey rsa:4096 -keyout tls.key -out tls.crt -days 365 -nod
 
-By default, Elasticsearch is using memory mapping (mmap) to efficiently access indices.
-Usually, default values for virtual address space on Linux distributions are too low for Elasticsearch to work properly, which may result in out of memory exceptions.
-To increase virtual memory ECK sets the recommended value by default.
+[id="{p}-pod-template"]
+=== Pod Template
+
+Pod templates are the same pod templates you know and love from stateful sets and deployments. You can provide your own to add new settings or merge settings with our defaults. For instance, if you want to add new labels to your pods, you can apply a pod template like so:
 
-A dedicated init container will set the kernel setting `vm.max_map_count=262144` on the host.
-This requires the init container to be privileged.
-This kernel setting can also be set on the host directly.
-In such case, you may disable the init container explicitly in the Elasticsearch specification:
 [source,yaml]
 ----
-spec:
-  setVmMaxMapCount: false
+    podTemplate:
+      metadata:
+        labels:
+          # additional labels for pods
+          foo: bar
 ----
 
-To get more info about this setting you can consult the Elasticsearch documentation:
-https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html
 
-Optionally, you can select a different type of file system implementation for the storage. Here you can find information about possible options:
-https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-store.html
+More common is setting resource requests and limits, which is covered in the section <<{p}-resource-requests-limits>>.
+
+You may want to also install additional plugins, which is described in the <<{p}-init-containers-plugin-downloads>> section.
+
+You may want to set environment variables to configure Elasticsearch, for instance to set the minimum and maximum heap size to `2g` and `4g` respectively you may modify the environment variables of the `elasticsearch` container as follows:
 
 [source,yaml]
 ----
-spec:
-  nodes:
-  - nodeCount: 3
-    config:
-      index.store.type: niofs
+    podTemplate:
+      spec:
+        containers:
+        - name: elasticsearch
+          env:
+          - name: ES_JAVA_OPTS
+            value: "-Xms2g -Xmx4g"
+----
+
+
+You can also refer to the Kubernetes documentation here for more information on pod templates:
+https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/#pod-templates[Pod Templates Overview]
+
+https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#podtemplatespec-v1-core[Pod Template Spec API Reference]
+
+
+[id="{p}-resource-requests-limits"]
+=== Resource Requests and Limits
+
+Resource requests and limits define how much RAM and CPU each individual container requires and how much they are allowed to use. You may want to set these in addition to the heap size as mentioned , though by default we will set the heap size to half of the container memory limit.
```

Comment:
> we will set the heap size to half of the container memory limit.
we don't do that anymore.
Probably worth syncing with @barkbay about https://github.com/elastic/cloud-on-k8s/pull/1252/files. It looks like both PR overlap here.

Reply:
Thanks, removed